packages/react-storage/src/components/StorageBrowser/tasks/useProcessTasks.ts

```diff
@@ -75,13 +83,22 @@ export const useProcessTasks: UseProcessTasks = <
 
   const createTask = React.useCallback(
     (data: T) => {
+      const getTask = () => tasksRef.current.get(data.id);
+      const { onTaskCancel, onTaskRemove } = callbacksRef.current;
+
       function remove() {
-        if (tasksRef.current.get(data.id)?.status === 'PENDING') return;
+        const task = getTask();
+        if (!task || task?.status === 'PENDING') return;
+        if (task && isFunction(onTaskRemove)) onTaskRemove(task);
+
         updateTask(data.id);
       }
 
       function cancel() {
-        if (tasksRef.current.get(data.id)?.status !== 'QUEUED') return;
+        const task = getTask();
+        if (!task || task?.status !== 'QUEUED') return;
+        if (task && isFunction(onTaskCancel)) onTaskCancel(task);
+
         updateTask(data.id, { cancel: undefined, status: 'CANCELED' });
       }
```

Comment:
Not doing it here, but feeling that `remove` can be dropped from the `Task` interface entirely and that `cancel` should receive a `remove?: boolean` argument to allow for removal of a `task` from the `tasks`